textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/common/RegexPredicate.java

```diff
@@ -36,5 +50,27 @@ public boolean satisfy(ITuple tuple) {
         }
         return false;
     }
+    
+    public List<Span> statisfySpan(ITuple tuple) {
+    	List<Span> res = new ArrayList<>();
+    	if (tuple == null) {
+    		return res; //empty array
+    	}
+    	IField field = tuple.getField(fieldName);

```

Comment:
@laisycs ... The current logic is correct for `TextField` and not for `StringField`. So change the existing `if` loop condition to `instanceof TextField`.
Write one more `if` block which handles the `StringField` instances. Here the key should match the field completely and no partial matching is allowed. Please check [Dictionary Matcher code](https://github.com/TextDB/textdb/blob/master/textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcher.java#L134) for reference.